### PR TITLE
get main and develop back in sync

### PR DESCRIPTION
**Pull request by iluise**
as described here https://github.com/clessig/atmorep/pull/53:
we merged https://github.com/clessig/atmorep/pull/37 into the main branch so to create a tag for the new single field runs. This PR puts `main` and `develop` back in sync. 
Since there are small conflicts but both `main` and `develop` are protected I am using `iluise/head` as intermediate branch, as it is the exact copy of `main` at the moment (please double check).

### CODE REVIEW

File: atmorep/core/evaluator.py [hunk 1]
```diff
@@ -155,11 +150,11 @@ def global_forecast( cf, model_id, model_epoch, devices, args = {}) :
     cf.batch_size_test = 24
     cf.num_loader_workers = 12 #1
     cf.log_test_num_ranks = 1
+
+    #TODO: temporary solution. Add support for batch_size > 1
```
Comment (clessig) [inline atmorep/core/evaluator.py:154]: Do we really have this restriction at the moment for the validation batch size?

File: atmorep/core/train.py [hunk 2]
```diff
@@ -56,13 +56,13 @@ def train_continue( wandb_id, epoch, Trainer, epoch_continue = -1) :
...
     cf.years_val = cf.years_test
-  
+ 
   # any parameter in cf can be overwritten when training is continued, e.g. we can increase the 
   # masking rate 
   # cf.fields = [ [ 'specific_humidity', [ 1, 2048, [ ], 0 ], 
   #                               [ 96, 105, 114, 123, 137 ], 
   #                               [12, 6, 12], [3, 9, 9], [0.5, 0.9, 0.1, 0.05] ] ]
-
+
```
Comment (clessig) [inline atmorep/core/train.py:65]: Can we remove this.

File: atmorep/core/train.py [hunk 3]
```diff
@@ -145,7 +141,7 @@ def train() :
   # training params
   cf.batch_size_validation = 1 #64
   cf.batch_size = 96
-  cf.num_epochs = 128
+  cf.num_epochs = 400 #128
```
Comment (clessig) [inline atmorep/core/train.py:144]: Do we need 400 here? 128 sounds more reasonable.

File: atmorep/core/train.py [hunk 4]
```diff
@@ -220,7 +217,7 @@ def train() :
   # # # cf.file_path = '/p/scratch/atmo-rep/data/era5_1deg/months/era5_y2021_res025_chunk8.zarr'
   # # cf.file_path = '/ec/res4/scratch/nacl/atmorep/era5_y2021_res025_chunk8_lat180_lon180.zarr'
   # # # cf.file_path = '/ec/res4/scratch/nacl/atmorep/era5_y2021_res025_chunk16.zarr'
-  cf.file_path = '/gpfs/scratch/ehpc03/era5_y2010_2021_res025_chunk8.zarr/'
+  cf.file_path = '/gpfs/scratch/ehpc03/era5_y1979_2021_res025_chunk8.zarr/'
```
Comment (clessig) [inline atmorep/core/train.py:220]: Could we make this easier to configure? Hard coding the path here is not nice.

File: atmorep/core/train_multi.py [hunk 5]
```diff
@@ -184,56 +197,81 @@ def train_multi() :
...
   cf.attention = False
 
   cf.rng_seed = None 
 
   # usually use %>wandb offline to switch to disable syncing with server
   cf.with_wandb = True
-  setup_wandb( cf.with_wandb, cf, par_rank, 'train-multi', mode='offline')  
+  setup_wandb( cf.with_wandb, cf, par_rank, 'train', mode='offline')  
+
+  # cf.file_path = '/p/scratch/atmo-rep/data/era5_1deg/months/era5_y2021_res100_chunk32.zarr'
```
Comment (clessig) [inline atmorep/core/train_multi.py:235]: The different commented out cases are not very nice, even less with the hard coded path.

File: atmorep/core/trainer.py [hunk 6]
```diff
@@ -186,7 +186,7 @@ def run( self, epoch = -1) :
       print( '{} : {} :: batch_size = {}, lr = {}'.format( epoch, tstr, cf.batch_size, lr) )
 
       self.train( epoch)
-
+
```
Comment (clessig) [inline atmorep/core/trainer.py:189]: Please remove

File: atmorep/core/evaluator.py (hunk 1 above)
Comment (iluise) [reply atmorep/core/evaluator.py:154]: there's a bug for validation batch sizes different from 1 I need to sit down and solve: https://github.com/clessig/atmorep/issues/33
But we agreed to wait for the forecast code first

**Comment by iluise** [comment]
okay, I pushed a new version that handles the hardcoded paths and the n_size calculation. in principle one just needs to change the path in setup.py before doing `pip install -e .`
Please take a look and let me know what you think. 
This is in principle the code that I want to merge in `epicure-dev` as well if it's okay for you.

**Review by clessig — APPROVED** [review]
Great, thanks!

**Review by clessig — APPROVED** [review]

**Pull request merged**